### PR TITLE
[DOC] Correct ReST syntax in "RocketClassifier"

### PR DESCRIPTION
**Pull request by rahulporuri**
#### Reference Issues/PRs

N/A

#### What does this implement/fix? Explain your changes.

This PR fixes the docstring in `RocketClassifier` to ensure that it gets rendered correctly. For the buggy rendering, please check the [RocketClassifier docstring](https://www.sktime.net/en/stable/api_reference/auto_generated/sktime.classification.kernel_based.RocketClassifier.html#sktime.classification.kernel_based.RocketClassifier). I chose to use the [ReST simple tables syntax](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#simple-tables), which is easier to read in code than the [ReST list table syntax](https://docutils.sourceforge.io/docs/ref/rst/directives.html#list-table) seen elsewhere

The correctly rendered docs on this branch - https://sktime--5564.org.readthedocs.build/en/5564/api_reference/auto_generated/sktime.classification.kernel_based.RocketClassifier.html#rocketclassifier

#### Does your contribution introduce a new dependency? If yes, which one?

No

#### What should a reviewer concentrate their feedback on?

Ensure that the docs built on this branch are rendered correctly

#### Did you add any tests for the change?

Not relevant

#### Any other comments?

We noticed a lot of similar issues with docstring rendering in the API docs. If contributions are welcome, we might slowly fix them in separate PRs.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/CONTRIBUTORS.md) with any new badges I've earned :-)
  How to: add yourself to the [all-contributors file](https://github.com/sktime/sktime/blob/main/.all-contributorsrc) in the `sktime` root directory (not the `CONTRIBUTORS.md`). Common badges: `code` - fixing a bug, or adding code logic. `doc` - writing or improving documentation or docstrings. `bug` - reporting or diagnosing a bug (get this plus `code` if you also fixed the bug in the PR).`maintenance` - CI, test framework, release.
  See here for [full badge reference](https://allcontributors.org/docs/en/emoji-key)
- [ ] Optionally, I've added myself and possibly others to the [CODEOWNERS](https://github.com/sktime/sktime/blob/main/CODEOWNERS) file - do this if you want to become the owner or maintainer of an estimator you added.
  See here for further details on the [algorithm maintainer role](https://www.sktime.net/en/latest/get_involved/governance.html#algorithm-maintainers).
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [ ] I've added the estimator to the API reference - in `docs/source/api_reference/taskname.rst`, follow the pattern.
- [ ] I've added one or more illustrative usage examples to the docstring, in a pydocstyle compliant `Examples` section.
- [ ] If the estimator relies on a soft dependency, I've set the `python_dependencies` tag and ensured
  dependency isolation, see the [estimator dependencies guide](https://www.sktime.net/en/latest/developer_guide/dependencies.html#adding-a-soft-dependency).

### CODE REVIEW

**Review by fkiraly — APPROVED** [review]
Thanks!

**Comment by rahulporuri** [comment]
sorry @fkiraly but the docs aren't rendering as expected. the issue looks deeper. let me change the PR back to a draft PR to prevent this from getting merged as is

**Review by fkiraly — CHANGES_REQUESTED** [review]
Yes, indeed - I should have checked on readthedocs, not in vs code which renders it correctly.

What you need to do to fix is adopt the same format as in `BaseTransformer.transform` (compare it to `BaseTransformer._transform` which still has a copy-paste of how `transform` looked like before it was corrected by @romanlutz)

The key is, this is rst and not markdown, so even if it renders in markdown it will not in rst, you need to use rst table syntax.

**Comment by rahulporuri** [comment]
@fkiraly thanks for the pointer. i ended up using ReST simple table syntax (I updated the PR description to reflect this). The API doc is now rendered correctly.

I see the same/similar issue exists in other places. I will try to open a separate PR to address them.

File: sktime/classification/kernel_based/_rocket_classifier.py
Comment (yarnabrina) [inline sktime/classification/kernel_based/_rocket_classifier.py:39]: For consistency, I suppose it should have a closing double quote.
```suggestion
    "minirocket"     "yes"               MiniRocketMultivariate
```

**Review by yarnabrina — CHANGES_REQUESTED** [review]
Thanks for your contribution.

I suggested a minor change below. Apart from that, please resolve the conflicts with `main` branch.

**Review by yarnabrina — APPROVED** [review]
Looks good to me 👍🏽

**Comment by rahulporuri** [comment]
@yarnabrina you might have to "dismiss review" for @fkiraly 's review requesting changes. i'm unable to merge the PR. Not sure what the general process is.

**Comment by fkiraly** [comment]
> Not sure what the general process is.

Core dev (not identical with proposer) merges after approving

**Review by fkiraly — APPROVED** [review]
Thanks!

**Comment by rahulporuri** [comment]
@fkiraly @yarnabrina a gentle reminder that this PR has been approved and is waiting for a merge

**Comment by yarnabrina** [comment]
There's a feature freeze announcement for PyData Global in our Discord. @fkiraly will probably merge this PR after it's over, shouldn't be more than 1/2 more days.

**Comment by fkiraly** [comment]
apologies, I was very busy recently, and the feature freeze got side tracked

**Pull request merged**
